src/umocktypes_c.c

```diff
@@ -24,15 +27,16 @@
         } \
         else \
         { \
-            size_t length = snprintf(NULL, 0, printf_specifier, *value); \
+            char temp[MAX_STRINGIFY_SIZE]; \
```

Comment:
>MAX_STRINGIFY_SIZE [](start = 22, length = 18)

I actually disklike that we would need to do this. The standard says that 

"The snprintf function is equivalent to fprintf, except that the output is written into
an array (specified by argument s) rather than to a stream. If n is zero, nothing is written,
and s may be a null pointer. Otherwise, output characters beyond the n-1st are
discarded rather than being written to the array, and a null character is written at the end
of the characters actually written into the array. If copying takes place between objects
that overlap, the behavior is undefined."

Notice the "If n is zero, nothing is written,
and s may be a null pointer."
I am quite surprised that gcc is unhappy with this.
What does unhappy mean? crash? warning?

Reply:
Some more information about this issue. MCU manufactories like NXP and Microchip provides optional libs for their own MCUs, for instance, NXP provides the Redlib, which is their "own (non-GNU) ISO C90 standard C library, with some C99 extensions". 

See more at https://community.nxp.com/thread/389152, and http://www.microchip.com/forums/m879660.aspx.

snprintf() is part of what they call C99 extensions, and they are probably not following the part of the c99 standard that accept null pointer.

At least for Redlib, umock-c works as a charm without this snprintf with null.